### ISSUE TITLE
Elementor v2.8 Beta 4 Release

### ISSUE THREAD

**Issue by shilo-ey** [elementor/elementor#9672]
===========

# [Beta 4 UPDATED - PLEASE READ BELOW!](https://github.com/elementor/elementor/issues/9672#issuecomment-562328620)

===========

Hi Elementors,

We are getting ready to release Elementor v2.8 and we would like to ask you to help us test Elementor v2.8 Beta, with 3 substantial design features:

- New color picker, with much better user experience in mind. 
- Dark mode, for better accessibility (or if you’re a goth). 
- Better responsive mode, making mobile web design even easier.

Before you test, please make sure you are using:

- Elementor Core v2.8 Beta
- Elementor Pro v2.7.3

## Meet The New Color Picker: Save Colors Instantly (#6968, #8982)

![picker](https://user-images.githubusercontent.com/45009641/69522246-968c9600-0f69-11ea-80b6-a145e6ac0f1b.gif)

Color is one of the most fundamental cornerstones of web design. If you’ve built Elementor sites before, you know that the color picker is a tool used dozens of times throughout each project.

It took a lot of work, but we completely rebuilt our color picker, and made it a lot more efficient.
Elementor had previously used the native WordPress “Iris” picker, a viable yet limited picker.

From now on, you can save your favorite colors straight from the picker itself. In fact, save colors on any picker, on any widget, on any page. It automatically syncs with every other picker on your site.

The new version, v2.8, brings a lightweight color picker, which from now on will allow saving your favorite colors simply by clicking on the “+” icon. 

By building this new Color Picker, we solved many other problems that our community raised on GitHub.

 
**Test flow:**

1. Go to a page you previously created or insert a new one from the Templates Library
2. Go to a widget which you added styles & color to
3. Click on Style tab and edit a previously populated color control
4. Check to see if your previously selected color is selected and present
5. Change the color, click on the “Add” (+) icon and see if the color is saved.
6. Close the color picker
7. Make sure the color you selected is present
8. Open the color picker and try deleting your recently saved color
9. Copy a color HEX / RGBA value (`#6a12e3` or `rgba(255, 66, 129, 0.6)`)
10. Drag in a new button widget, Go to Style tab and open the color picker
11. Paste the copied color to the input field and see if the color updated
12. Make sure everything works as expected on Frontend.

 
## Introducing Dark Mode (#9263, #8249, #9125)

![Dark Mode](https://user-images.githubusercontent.com/45009641/69496136-7f9d6380-0ed7-11ea-8a0e-220ae223cb83.gif)

Tired of getting sore eyes when editing your websites until the wee hours of the night? Meet Elementor Dark Mode!
Save your laptops’ battery life, while also saving the environment. Enjoy contrast improvement in the editor panel for better accessibility, or simply make your Editor Panel better fit your dark eighties music preference. I recommend listening to [A Forest by The Cure.](https://www.youtube.com/watch?v=RGT4V6JmINA)  🖤🥀
 
**Test flow:**

1. Edit a page or create a new page with Elementor
2. In the Editor panel click on the hamburger menu > Preferences
3. Under “UI Theme” choose “Dark”
4. Make sure Dark mode is visible
5. Navigate throughout your entire Editor and see that everything looks as expected
6. Go back to step (2) and under “UI Theme” select “Auto Detect”
7. Go to your OS preferences and set it to “Dark”
8. Make sure Dark mode is visible

## Better Control Over Responsive Modes 
 
In Elementor v2.8, we added two new responsive features: We added a clear indication showing which elements you hid from any responsive view, and also added a new switcher when switching between desktop, tablet, and mobile mode. 

### Better Responsive Visibility Indication (#8020, #4775)

![Better Responsive Visibility Indication ](https://user-images.githubusercontent.com/45009641/69496143-9a6fd800-0ed7-11ea-8b74-d655f63b0f4e.gif)

One of the feedback we commonly get is “Why is my HIDDEN element still visible in the editor preview?! 😡”
We feel your anger… From now on, you can cross off that question.
When hiding an element, and switching to the view it’s hidden on, you will see that this element is “muted”.

**Test flow:**

1. Create a new page with Elementor
2. Open the Templates Library, and connect your Elementor account for verification
3. Import a full-page Template from the library
4. Select a section, go to Advanced tab > Responsive > and switch “Hide on Tablet” on
5. Go to Tablet view using the Editor bottom bar
6. Make sure everything works as expected and you can see the “muted” version of that hidden section
7. Close the panel by clicking on the panel collapse button and make sure the “muted” element disappeared

## New: Responsive Mode Switcher Control

![Responsive Mode Switcher Control ](https://user-images.githubusercontent.com/45009641/69496150-ae1b3e80-0ed7-11ea-802d-e8ba61e975af.gif)

You know how when you switch from mobile to desktop, the switcher runs horizontally? This may have not crossed your mind, but that actually limits us when thinking about extending the switcher. This is why we turned the switcher to a vertical one. **It is one more step towards one of our most requested features regarding breakpoints**.

**Test flow:**

3. Edit a page or create a new page with Elementor
4. Drag in a button widget
5. Go to Alignment control and play around with the responsive picker
6. Switch to Mobile view, Tablet view and back to Desktop view
7. Make sure everything works as expected
8. Try using this control in different browsers and instances.

## Compatibility with WordPress v5.3 
WordPress 5.3 (“Kirk”) is officially [here](https://wordpress.org/support/wordpress-version/version-5-3/) and incorporates various changes and improvements. In this version, we made sure everything will continue working smoothly with Elementor, and won’t hold you back when creating your site.

**Test flow:**

1. Prior to updating to WP v5.3, install Elementor v2.8 Beta
2. Create a new page with Elementor
3. Open the Templates Library, and connect your Elementor account for verification
4. Import a full-page Template from the library
5. Upgrade WordPress to v5.3
6. Explore the dashboard and make sure everything works and looks as expected
7. Go to your recently created page and make sure everything stayed the same
8. Explore the Editor Panel to see if there are any glitches

## Connect Your Free Elementor Account to Access Template Library

The Elementor Library was one of the first features we released and has grown to become one of our most substantial features. Its functionality has expanded to include block and page templates, popups, dynamic templates and more.

On our continuous efforts to improve this feature, starting from version 2.8, we have added infrastructure for a smarter library that incorporates remote access, login capabilities, and more personalized experience in the future.

This will enable us to take the Template Library to the next level. Not just a place of storage for templates. The Library Login will enable us to add valuable personalization features such as a ‘favorites’ and ‘recently used’ views, so you can have easy access to the templates you really want.

As a user, you will gain better control over who uses your account and website and get added security. It will also help us keep the copyrights of some of our proprietary assets, including stock images and designer-made visuals.

**Test Flow:**

1. Immediately after updating Elementor to v2.8, Create a new page
2. A modal will appear, prompting you to Connect. For the time being, go ahead and close this modal
3. Open the Templates Library and insert a template
4. You will see a message prompting you to Connect 
5. Click “Connect to Insert”
6. Connect your Elementor account or sign up for a new free account
7. Approve the connection and make sure the template is inserted properly

## Under the Hood Improvements:
As was in the previous version, in the following releases, we will gradually release major infrastructure improvements, which will allow Elementor to be more extendable, reliable and stable. Those improvements include more implementations to the new Editor JS API mechanism, which allows us to better handle the action history (Say nice hello to page settings history).

Learn more about [JS API Commands](https://developers.elementor.com/js-api/js-api-commands/), [JS API Routes](https://developers.elementor.com/js-api/js-api-routes/) and see the [v2.8 full deprecation list](https://developers.elementor.com/v2-8-0-planned-deprecations/).

**Test flow:**

These features require a sanity test for the entire system.
1. Create a page and add widgets to it, hit publish and make sure the flow is as usual.
2. Create a section and save it as a template. Try to insert it into your previously created pages and see if everything works as expected.
3. Try exploring the redo, undo and revisions options in different document types (Popups, Headers, Global widget, etc...) and see if everything works as expected.
4. Pro users: create a single template, import content from the library, set conditions and make sure everything works as expected.

--------
## More Improvements

- Tweak: Added `background-size` and `background-position` controls to Background Slideshow (#9199)
- Tweak: Removed Google+ network from “Recommended” tab in Social Icons widget (#9393)
- Tweak: Added dynamic capabilities to alert widget (#9436)
- Tweak: Added "Play on Mobile" option to allow iOS autoplay to Video widget (#3442)
- Tweak: Added Icon control inline skin
- Tweak: Delete revisions option is no longer supported since WP 5.3 ([Topic](http://core.trac.wordpress.org/ticket/43709))
- Tweak: Added `elementor/settings/controls/checkbox_list_cpt/post_type_objects` filter to control over displayed CPT in WordPress Dashboard UI (thanks @sc0ttkclark)
- Tweak: Moved “Editing Handles” and “Lightbox in editor” options to the new “Preferences” section
- Tweak: Added compatibility with WordPress v5.3 
- Tweak: Updated E-gallery library to v1.1.0
- Tweak: Added `playsinline` attribute to Vimeo and YouTube sources to allow autoplay on iOS devices in Background Video
- Tweak: Added Documents count to System Info
- Tweak: Added global utility function for adding link attributes in Button widget
- Tweak: Re-designed alerts and notices in the Editor Panel
- Tweak: Changed “Custom Positioning” section label to “Positioning”
- Tweak: Replaced default Google+ network icon with YouTube icon in Social Icons widget
- Tweak: Improved styling of checkboxes and radio inputs
- Tweak: Lined up Elementor system buttons styling
- Tweak: Allow access to Templates Library for connected users only
- Tweak: Changed icon control skin to “Inline” in Toggle and Accordion widgets
- Tweak: Added Icon Manager “Recommended” tab in Toggle and Accordion widgets
- Tweak: Changed all occurrences of `text-overflow: ellipsis` to a single `mixin`
- Tweak: Added “Pause On Interaction” option to Image Carousel widget
- Tweak: Dropped Support for WP v4.9, now supporting WP v5.0+
- Fix: Use CSS Flex to align and order button icon (#561)
- Fix: Remove Ninja Forms compatibility script which causes server issues (#8267, Developers Blog Post)
- Fix: Added `title` attribute to embedded iframes (#9374, thanks @shipley-dcc)
- Fix: Reset style in a button doesn't work on colors (#8454)
- Fix: No fallback when SVG file doesn’t exist (#9481)
- Fix: Adding a link overrides the Title and Description colors in Testimonial widget
- Fix: Added better output escaping in the attributes for checkbox lists in Admin forms (thanks @sc0ttkclark)
- Fix: “Delete All Content” doesn't close current widget editing options
- Fix: Pause on hover option doesn't work in Image Carousel widget
- Fix: Hide Beta Testers signup form if the user already subscribed
- Fix: Recommended tab icons don’t change when there’s more than one `Icons` control per widget
- Fix: Transition Duration option not applying to elements background
- Fix: Hidden sticky section generates redundant scroll offset when using Anchors
- Fix: Inserting a saved section as a template uses the same `ID`
- Deprecated: See all deprecations to this version in our [Developers Deprecations Post](https://developers.elementor.com/v2-8-0-planned-deprecations/)

 
-------

## How to Install the Beta Version
If you still haven't enabled beta testing in Elementor, please follow these steps:

- Go to Elementor > Tools > Versions.
- Enable the beta tester feature.
- Go to the Dashboard > Updates and update to the latest beta version.
- You can also download the new version here: [downloads.wordpress.org/plugin/elementor.2.8.0-beta4.zip](https://downloads.wordpress.org/plugin/elementor.2.8.0-beta4.zip)

## Notes

For bugs and remarks, please reply with a comment on this issue here on GitHub.

Please keep this related to the Current beta Only! any other issue or feature request, feel free to open a new issue.

Designated time for release: `December 9, 2019` (Estimated)

**Many thanks for your support and help!**

Cheers,
Elementor Team

**Comment by rodolphebertozzo** [comment]
Hello, it's a very good job !!
One issue : The colours inserted in the different pickeds are not saved

**Comment by rodolphebertozzo** [comment]
Hello, it's a very good job !!
One issue : The colours inserted in the different pickeds are not saved
![bug elementor](https://user-images.githubusercontent.com/56270626/69498674-c6e11f80-0eea-11ea-8337-ef6fb545fbfc.PNG)

**Comment by fabianenzensperger** [comment]
> Hello, it's a very good job !!
> One issue : The colours inserted in the different pickeds are not saved
> ![bug elementor](https://user-images.githubusercontent.com/56270626/69498674-c6e11f80-0eea-11ea-8337-ef6fb545fbfc.PNG)

I do not have this problem. With me, all colors are stored correctly via the + symbol and are available everywhere.

**Comment by fabianenzensperger** [comment]
Hi, thanks for the beta and the soon to be released Elementor (Pro also?) Version 2.8.

It's all fine, but I have a concern. An important and good step in the right direction is done with the new behavior of responsive content in edit mode. However, it is so much better for me if the elements are not displayed at all.

For example, if I have different items in the menu, some of which are not displayed on the desktop and in the mobile, they will be displayed in Edit mode and will not show the available space as it should, and these items will interfere with mobile operation very.

Is there really no way to hide these elements completely? That would be THAT feature that I wish...

**Comment by fabianenzensperger** [comment]
The new responsive Mode Switcher Control are very helpful and the darkmode is perfect for apples "new" darkmode. It looks all correctly so far. :)

**Comment by black-eye** [comment]
Hi, I gave it a quick try and here are some flash observations:

1. Tweak: Moved “Editing Handles” to the new “Preferences” section
It doesn't matter to me where the settings are (although the former place was easier for me to find), but the problem is when the setting isn't kept accross the upgrades. Previously I have the Editing Handles turned on, after the update they were gone again.

2. Color Picker and Central Color Palette 
New picker isn't compatible with [Central Color Palette plugin](https://wordpress.org/plugins/kt-tinymce-color-grid/). I'm aware the problem is on their side, so I've opened the issue for it in their WP support page: https://wordpress.org/support/topic/compatibility-with-the-new-elementors-color-picker/. This is just to let you know.

3. Other Color Picker little bugs

3.1. When I open the Picker with no color and decide to close it down without choosing any color, I would expect, the color stays unselected. Unfortunately, the most of the time pure black is selected instead.

3.2. When I'm too quick with the closing the Picker right after adding a new color, the Picker stops working correctly - I'm not able to add new color anymore, nor I'm able to delete colors. In FireFox the tooltip "Add Picked Color" doesn't disapear and stays there even after closing the picker.

Both issues you can see in action in this video: https://drive.google.com/file/d/14v7ReFGMtu6nNI8ngYpLxVJsZJXhlD1J/view

4. It would be nice to be able to reorder the colors in the picker manually.

Thanks

**Comment by drawcard** [comment]
My immediate impression of the "responsive hidden item" feature is that, if you're working on a site that happens to use greyscale images, it isn't very apparent that this element is hidden amongst everything else. It's not common but it could happen, especially on photography portfolio sites.

Perhaps a solution is to overlay an icon as well, to make it obvious to the viewer that this panel is not visible? This would also make the feature super accessible to those with colour blindness.

As a bonus, you could add dynamic text to reflect the hidden statuses across "Desktop", "Tablet" and "Mobile" -  shortened to D, T or M below the icon. That would save the user an extra click to see the visibility status of that item.

**Mockup:**

![image](https://user-images.githubusercontent.com/1435638/69514274-a375ad80-0f9a-11ea-88e9-210e3d6b31ca.png)

**Comment by black-eye** [comment]
@shilo-ey Just a quick question: you are mentioning 2.8. is WP 5.3 compatibile. Does it mean earlier versions aren't? Should we avoid updating to 5.3 before E 2.8 is out?

I've updated several sites already, with no visible issues, but I didn't check them very deeply.

**Comment by pingram3541** [comment]
This doesn't appear to be Beta only but might be a good additional fix for this beta if we can slip it in.  I tested on two sites, one w/ beta and one without.

I am encountering some strange behavior as it pertains to icons.  If I edit an icon element and decide I want to undo my changes using the keyboard shortcut CTRL/CMD + Z, the icon itself resets to the Star icon and there is no way to undo it.

The change is also not reflected in history.  One must manually seek through the icon library again and choose the original icon.

Additionally, one doesn't have to make any changes for this to take effect, simply click on any existing icon element and press CTRL/CMD + Z and see the icon revert to the Star icon with any history events shown.

**Comment by shilo-ey** [comment]
[**@fabianenzensperger**](https://github.com/fabianenzensperger) commented on [Nov 24, 2019, 11:04 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-557927867 "2019-11-24T21:04:55Z - Replied by Github Reply Comments"):
> The real question is, how do I change the fixed color without having to delete the "old" and create a "new one"? Also in relation to the colors on the previous website, which I do not want to recolor all.

The Saved Colors feature isn't the Global Colors. That means that it uses as shortcuts to the color palette.




[**@fabianenzensperger**](https://github.com/fabianenzensperger) commented on [Nov 24, 2019, 11:12 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-557928532 "2019-11-24T21:12:37Z - Replied by Github Reply Comments"):
> Hi, thanks for the beta and the soon to be released Elementor (Pro also?) Version 2.8.
> 
> It's all fine, but I have a concern. An important and good step in the right direction is done with the new behavior of responsive content in edit mode. However, it is so much better for me if the elements are not displayed at all.
> 
> For example, if I have different items in the menu, some of which are not displayed on the desktop and in the mobile, they will be displayed in Edit mode and will not show the available space as it should, and these items will interfere with mobile operation very.
> 
> Is there really no way to hide these elements completely? That would be THAT feature that I wish...

We plan to hide those elements completely, but in order to make the transition gracefully, we need to make these changes incrementally.
This is in order to promote and cement new behavior and work habits



[**@black-eye**](https://github.com/black-eye) commented on [Nov 24, 2019, 11:27 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-557929744 "2019-11-24T21:27:22Z - Replied by Github Reply Comments"):
> Hi, I gave it a quick try and here are some flash observations:
> 
> 1. Tweak: Moved “Editing Handles” to the new “Preferences” section
> It doesn't matter to me where the settings are (although the former place was easier for me to find), but the problem is when the setting isn't kept accross the upgrades. Previously I have the Editing Handles turned on, after the update they were gone again.

You are right, until v2.8 this setting was controlled site-wide. from now on, each user can select his preference of use.
This is why this is a different setting.

> 2. Color Picker and Central Color Palette 
> New picker isn't compatible with [Central Color Palette plugin](https://wordpress.org/plugins/kt-tinymce-color-grid/). I'm aware the problem is on their side, so I've opened the issue for it in their WP support page: https://wordpress.org/support/topic/compatibility-with-the-new-elementors-color-picker/. This is just to let you know.

Thanks!

> 3. Other Color Picker little bugs
> 
> 3.1. When I open the Picker with no color and decide to close it down without choosing any color, I would expect, the color stays unselected. Unfortunately, the most of the time pure black is selected instead.

Thanks, will be fixed in beta2

> 3.2. When I'm too quick with the closing the Picker right after adding a new color, the Picker stops working correctly - I'm not able to add new color anymore, nor I'm able to delete colors. In FireFox the tooltip "Add Picked Color" doesn't disapear and stays there even after closing the picker.

Thanks, will be fixed in beta2


> 4. It would be nice to be able to reorder the colors in the picker manually.
> 
> Thanks

We will consider adding it,

**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

Thanks again to everyone.

**Comment by mskks64** [comment]
Very difficult to read some of the custom css colors in the dark mode. 

![Screen Shot 2019-11-25 at 9 59 18 AM](https://user-images.githubusercontent.com/11357379/69556475-61347800-0f6a-11ea-8b70-b8dd98bde7ad.png)

**Comment by pingram3541** [comment]
Good point @mskks64 maybe time to introduce editor themes.

Being able to theme the css editor (both ACE and CodeMirror support themes) by choosing from a drop down in the back end would be a nice addition to compliment dark/light modes and they don't add too much weight but a small amount of css.

**Comment by black-eye** [comment]
> > 1. Tweak: Moved “Editing Handles” to the new “Preferences” section
> >    It doesn't matter to me where the settings are (although the former place was easier for me to find), but the problem is when the setting isn't kept accross the upgrades. Previously I have the Editing Handles turned on, after the update they were gone again.
> 
> You are right, until v2.8 this setting was controlled site-wide. from now on, each user can select his preference of use.
> This is why this is a different setting.

@shilo-ey OK, got it. But could it be possible to keep the old way too? To have the global setting in the old place in the dashboard and this new one for custom override right in the editor?

I use these handles quite often and so do my clients/editors. On some sites there are many editors and I would have to reconfigure those presets manually one by one. Most of the users wouldn't be able to find the right place by themselves, not to mention some of them don't have the full editor rights (Which is BTW another point - could those editors with limited rights have the way to turn this option on? - I have to try it.)

**Comment by rodolphebertozzo** [comment]
There is also a small display problem in the right-click menu
![beta 2](https://user-images.githubusercontent.com/56270626/69566157-bdbd8480-0fb6-11ea-83cf-41b9f9f156dc.PNG)

**Comment by Arilliya** [comment]
Not sure if this is something I missed or not added, but I don't currently see a way to remove added colors. This is a big must.

**Comment by drawcard** [comment]
> We plan to hide those elements completely, but in order to make the transition gracefully, we need to make these changes incrementally.
This is in order to promote and cement new behavior and work habits

I am a bit concerned about this @shilo-ey - what happens in the following scenarios?

- User hides the element in desktop mode, causing it to disappear in the regular editor
- User hides the element in all three breakpoints accidentally, and the element completely disappears (it is now only accessible by digging through the Navigator panel)
- A user hides many items across the site,  forgets they are hidden, then has no reference point to retrieve or edit them 
- Other weird interactions like nested items  disappearing if a section is hidden.

To me it seems like a bad move to have widgets completely disappear in the editor, without some kind of visual indicator or representation that they still exist. Doesn't live preview do enough with showing the finished results? 

Thanks for your consideration.

**Comment by parfilov** [comment]
Hello! Colorpicker is unusual but cool ))
I see strange sticky "effect" on transparency bar + mouse cursor. Se here https://youtu.be/Vls6H7dSybQ

**Comment by parfilov** [comment]
Oddities with hex and rgba. Some time when go to the piker, color is in rgba, the next time — 6 or 8 digits Hex.

**Comment by sprad001** [comment]
The color picker ads color but I do not see how I can delete saved colors.
I had an issue when adding a new color the tool tip that says "Add Picked Color" gets stuck and remains on the screen even after the color picker is closed.
![image](https://user-images.githubusercontent.com/58193070/69583015-99e55780-1025-11ea-9a37-1060de804ebe.png)
I experimented on one of my existing pages with Icon Lists and set the icon colors. I wanted to transfer the same color and style to other icon list with the Copy and Paste Style. This pasting did not work, the previous color of the icons remained. Then I tried to edit the color on the Icon List where the Paste Style did not work and found that the color changes became irresponsive. I could change the color in the color picker but that did not reflect on the icon color, it remained unchanged on the icon list where I pasted the style.
Here is the interesting part: I deleted this color irresponsive Icon List and went to the History and stepped back to undo the delete. The icon list came back with the correctly applied color that was pasted the style on it.

**Comment by webcolor-in** [comment]
Problem: Color picker changes to black color when we click on color button twice (open/close popup)
Steps to reproduce: Click on color thumb to open color popup. Don't choose any color and click on color button so popup closes. Now if you had no color selection it changes color to Black.
Expected: It should not alter color unless we select any other color.

**Comment by shilo-ey** [comment]
[**@rodolphebertozzo**](https://github.com/rodolphebertozzo) commented on [Nov 25, 2019, 8:06 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-558273195 "2019-11-25T18:06:58Z - Replied by Github Reply Comments"):
> There is also a small display problem in the right-click menu
> ![beta 2](https://user-images.githubusercontent.com/56270626/69566157-bdbd8480-0fb6-11ea-83cf-41b9f9f156dc.PNG)

Which browser? OS? try clearing your cache and retry.


[**@Arilliya**](https://github.com/Arilliya) commented on [Nov 25, 2019, 10:35 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-558328724 "2019-11-25T20:35:04Z - Replied by Github Reply Comments"):
> Not sure if this is something I missed or not added, but I don't currently see a way to remove added colors. This is a big must.

Try dragging the colors. How do you feel regarding this behavior?

[**@Drawcard**](https://github.com/Drawcard) commented on [Nov 25, 2019, 11:13 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-558342351 "2019-11-25T21:13:23Z - Replied by Github Reply Comments"):
> > We plan to hide those elements completely, but in order to make the transition gracefully, we need to make these changes incrementally.
> This is in order to promote and cement new behavior and work habits
> 
> I am a bit concerned about this @shilo-ey - what happens in the following scenarios?
> 
> - User hides the element in desktop mode, causing it to disappear in the regular editor
> - User hides the element in all three breakpoints accidentally, and the element completely disappears (it is now only accessible by digging through the Navigator panel)
> - A user hides many items across the site,  forgets they are hidden, then has no reference point to retrieve or edit them 
> - Other weird interactions like nested items disappearing if a section is hidden.
> 
> To me it seems like a bad move to have widgets completely disappear in the editor, without some kind of visual indicator or representation that they still exist. Doesn't live preview do enough with showing the finished results? 
> 
> Thanks for your consideration.

Don't be :) we won't harm the way users work and we will make sure everything will still be clear to comprehend.


[**@sprad001**](https://github.com/sprad001) commented on [Nov 26, 2019, 12:32 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-558370621 "2019-11-25T22:32:20Z - Replied by Github Reply Comments"):
> I did this test in Firefox.
> The color picker ads color but I do not see how I can delete saved colors.
> I had an issue when adding a new color the tool tip that says "Add Picked Color" gets stuck and remains on the screen even after the color picker is closed or even if you go to edit something else.

Thanks, will be fixed in beta2

[**@parfilov**](https://github.com/parfilov) commented on [Nov 25, 2019, 11:37 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-558351216 "2019-11-25T21:37:11Z - Replied by Github Reply Comments"):
> Oddities with hex and rgba. Some time when go to the piker (the same element of the same widget), color is in rgba, the next time — 6 or 8 digits Hex.

Thanks,

The picker works in a way that HEX/HEXA values are the default display value, and for better compatibility and support, when you set alpha to the color, we convert those to rgba values.

If no alpha is set - you can see that it remains HEX

The picker accepts HEX, HEXA, HSL, and RGBA
 


> I experimented on one of my existing pages with Icon Lists and set the icon colors. I wanted to transfer the same color and style I just modified with the new color picker to other icon list with the Copy and Paste Style. This pasting did not work, the previous color of the icons remained. Then I tried to edit the color on the Icon List where the Paste Style did not work and found that the color changes became irresponsive. I could change the color in the color picker but that did not reflect on the icon color, it remained unchanged on the icon list where I pasted the style.
> Here is the interesting part: I deleted this color irresponsive Icon List and went to the History and stepped back to undo the delete. The icon list came back with the correctly applied color that was pasted the style on it before

Can't reproduce, please make sure your cache is cleared and you don't have any console errors.

**Comment by drawcard** [comment]
> Don't be :) we won't harm the way users work and we will make sure everything will still be clear to comprehend.

@shilo-ey thanks for responding - I am sure you will make the best decision :) For v2.8 If you could please consider [adding a visual icon](https://github.com/elementor/elementor/issues/9672#issuecomment-557993800) on top of the greyscale tint (for better accessibility) that would be a great step forward.

**Comment by rodolphebertozzo** [comment]
@shilo-ey Despite the empty cache and a browser change, the problem with the right-click menu is still there
![beta 2](https://user-images.githubusercontent.com/56270626/69626933-eb9fd900-1049-11ea-8d6b-a7bc29358dd7.PNG)

**Comment by godvsgodvs** [comment]
With the new color picker I actually lose functionality. I use the Central Color Palette plugin which gave me the same colors for Elementor and also the Page Builder Framework theme which I use for font colors. This update breaks that and actually locks up the Elementor editor if it's enabled. So now I would have one color palette for page editing and another for global fonts. Not cool.

**Comment by shilo-ey** [comment]
Hi all!

We just released a new beta version, please update your version to v2.8.0 Beta2.

This beta release includes:

- Tweak: Redesigned delete colors method in Color Picker
![new](https://user-images.githubusercontent.com/45009641/69657638-515c8700-1083-11ea-9d47-7290cfe9ebaf.gif)
- Tweak: Added `inline` support for choose-like controls
- Fix: Connect splash screen style glitches
- Fix: Hide the “Add New Color” tooltip in Firefox after interaction in Color Picker
- Fix: `#000000` value being set when not closing an empty Color Picker
- Fix: On first Color Picker hue selector click the selected color doesn't update
- Fix: Multiple Color Picker controls in same section prevents saving colors beyond the first one
- Fix: Double rendering of elements in Editor
- Fix: Some keyboard shortcuts not working


Please share your thoughts regarding the new delete method and let us know which one do you like better.


**Please keep this related to the Current beta Only! any other Issue Report or Feature Request feel free to open a new issue.**

Thanks again to everyone.

**Comment by cavalierlife** [comment]
I like the new delete method better. Two thumbs up.

**Comment by fabianenzensperger** [comment]
> Hi all!
> 
> We just released a new beta version, please update your version to v2.8.0 Beta2.
> 
> This beta release includes:
> 
> * Tweak: Redesigned delete colors method in Color Picker
>   ![new](https://user-images.githubusercontent.com/45009641/69657638-515c8700-1083-11ea-9d47-7290cfe9ebaf.gif)
> * Tweak: Added `inline` support for choose-like controls
> * Fix: Connect splash screen style glitches
> * Fix: Hide the “Add New Color” tooltip in Firefox after interaction in Color Picker
> * Fix: `#000000` value being set when not closing an empty Color Picker
> * Fix: On first Color Picker hue selector click the selected color doesn't update
> * Fix: Multiple Color Picker controls in same section prevents saving colors beyond the first one
> * Fix: Double rendering of elements in Editor
> * Fix: Some keyboard shortcuts not working
> 
> Please share your thoughts regarding the new delete method and let us know which one do you like better.
> 
> **Please keep this related to the Current beta Only! any other Issue Report or Feature Request feel free to open a new issue.**
> 
> Thanks again to everyone.

Much better, perfect! :)

**Comment by black-eye** [comment]
> Please share your thoughts regarding the new delete method and let us know which one do you like better.

Yeah, the new delete method is definitely better. The old way would have had some meaning, if it could be possible to rearange the swatches by dragging them around. But without this feature, this way is much clearer.

Moreover, I really like the "side-effect feature" it produces - the trash bin shows me, when I coincidentally hit a saved color. Before, I actually didn't even notice, the active swatch is marked with this tiny little border, which is practically invisible. It should be much bolder or few pixels indented. Or (even better) labeled with a little check mark inside.

I also appreciate the unselected color is now trasparent again and not gray, which could cause confusion wheather the color is really unselected or just gray.

**Comment by drawcard** [comment]
Small point re: colour picker - can we have the HEX / RGBA values in `monospace` font? an 0 looks like an O otherwise 😁

**Comment by sprad001** [comment]
**I actually like the drag color delete feature on the color picker**. Less chance of accidentally deleting a color. The problem was with the drag color delete feature that it was not mentioned how to use it in the email and it was not obvious. Perhaps implement both and have it selectable in the settings and maybe a tooltip showing up mentioning the drag delete feature.

**Comment by ocbroadband** [comment]
Appears that an issue was introduced into 2.8 B2 regarding the archive posts widget.  I can drag it into the page, I get the Content / Style / Advanced options in the menu, but I am unable to click on them.
![image](https://user-images.githubusercontent.com/13633011/69696465-9dfc8e00-1094-11ea-84d0-693bb6546887.png)
If I attempt to click on any of them, or attempt to edit via the handles, the menu items disappear altogether.

**Comment by black-eye** [comment]
> **I actually like the drag color delete feature on the color picker**. Less chance of accidentally deleting a color. 

This is actually a good point. The delete button should at least have a red color, be wider or something like that. And as I wrote earlier - if the swatches were dragable, it would be a different story...

**Comment by srcek** [comment]
There seem to be some glitches with the color picker. No selected color shows as grey which is confusing. If I clear the color or remove the value sometimes it reappears when I click on the picker. Intention is to clear any custom color values and make the element use default global ones.

**Comment by rodolphebertozzo** [comment]
Now that the 2 color suppression features have been developed, it would be interesting to leave the choice to the user in Elementor's preferences so that he can choose the method that suits him best. It would be nice to be able to move the picked ones around, to move them left or right. Would it also be possible to add an icon to the alert widget, next to the alert title, an example: 
![exemple](https://user-images.githubusercontent.com/56270626/69738281-25e6a480-1136-11ea-9947-3249b0a6b51f.png)

**Comment by mskks64** [comment]
Changing column width by typing in the number causes it to only move to the first digit (i.e. 75% is only doing 7%.

<img width="1274" alt="Screen Shot 2019-11-27 at 4 22 41 PM" src="https://user-images.githubusercontent.com/11357379/69763807-ff276e80-1133-11ea-9a36-5c10f2b689fb.png">
<img width="295" alt="Screen Shot 2019-11-27 at 4 23 05 PM" src="https://user-images.githubusercontent.com/11357379/69763809-ffc00500-1133-11ea-8dae-8ccb787cdc26.png">

**Comment by mskks64** [comment]
Another issue, dragging column to the correct size is saving everything in the history panel
<img width="281" alt="Screen Shot 2019-11-27 at 4 32 56 PM" src="https://user-images.githubusercontent.com/11357379/69763850-1b2b1000-1134-11ea-8475-09b19c829ae0.png">

**Comment by sprad001** [comment]
I reported with Beta 1 that for me the Copy and Paste Style did not work. Unfortunately that problem still exists for me in Beta 2 and I did all the usual clear cache and regenerate all CSS wherever it is possible.

**Comment by shilo-ey** [comment]
[**@srcek**](https://github.com/srcek) commented on [Nov 27, 2019, 3:08 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-559080210 "2019-11-27T13:08:26Z - Replied by Github Reply Comments"):
> There seem to be some glitches with the color picker. No selected color shows as grey which is confusing. If I clear the color or remove the value sometimes it reappears when I click on the picker. Intention is to clear any custom color values and make the element use default global ones.

Those issues fixed on beta 2, please try to update and clear cache.


[**@mskks64**](https://github.com/mskks64) commented on [Nov 28, 2019, 12:36 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-559273792 "2019-11-27T22:36:06Z - Replied by Github Reply Comments"):
> Changing column width by typing in the number causes it to only move to the first digit (i.e. 75% is only doing 7%.

Can't reproduce this issue. 


[**@mskks64**](https://github.com/mskks64) commented on [Nov 28, 2019, 12:36 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-559273985 "2019-11-27T22:36:53Z - Replied by Github Reply Comments"):
> Another issue, dragging column to the correct size is saving everything in the history panel

Good catch! thanks.


[**@sprad001**](https://github.com/sprad001) commented on [Nov 28, 2019, 3:16 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-559303198 "2019-11-28T01:16:37Z - Replied by Github Reply Comments"):
> I reported with Beta 1 that for me the Copy and Paste Style did not work. Unfortunately that problem still exists for me in Beta 2 and I did all the usual clear cache and regenerate all CSS wherever it is possible.

Now managed to understand the scenario and reproduce it. will be fixed. thanks


**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

Thanks again to everyone.

**Comment by rodolphebertozzo** [comment]
Would it also be possible to add an icon to the alert widget, next to the alert title, an example:
![exemple](https://user-images.githubusercontent.com/56270626/69824393-e7bbb480-120b-11ea-8219-039edae95c8c.png)

**Comment by shilo-ey** [comment]
[**@rodolphebertozzo**](https://github.com/rodolphebertozzo) commented on [Nov 28, 2019, 7:21 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-559566682 "2019-11-28T17:21:36Z - Replied by Github Reply Comments"):
> Would it also be possible to add an icon to the alert widget, next to the alert title, an example:
> ![exemple](https://user-images.githubusercontent.com/56270626/69824393-e7bbb480-120b-11ea-8219-039edae95c8c.png)

As mentioned:
**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

**Comment by rodolphebertozzo** [comment]
Also remember to add the dark mode here ;)
![modele](https://user-images.githubusercontent.com/56270626/69912856-a9064400-142f-11ea-98a8-dfaa13d102fd.PNG)

**Comment by knireis** [comment]
Tweak: Moved “Editing Handles” options to the new “Preferences” section

In 2.8 it does not take the previous setting from 2.7. It just resets itself to the default 'no'. Can you change this so it takes previous setting in account?

**Comment by black-eye** [comment]
@knireis 
> 
> Tweak: Moved “Editing Handles” options to the new “Preferences” section
> 
> In 2.8 it does not take the previous setting from 2.7. It just resets itself to the default 'no'. Can you change this so it takes previous setting in account?

Yes pls, preserve the previous settings as default. I've post the same request 6 days ago.

**Comment by black-eye** [comment]
One more question:
How exactly should the Library connection work accross several WP accounts?

1. I have Pro acount connected to the email for the main WP Admin account.
2. This account seems to be connected to library (at least it's said so on page Elementor > Connect)
3. For the first time, when I tried to insert a block from a library, the connection error was thrown.
4. After a refresh (which I did as suggested by error message) I was requested to connect the library to an account again.
5. After a connection was established, I was finaly able to insert the block from the library
6. Then I switched the WP account for an Editor user.
7. When I opened a page for editing I was requested to connect to library again (even before the page was opened).
8. When I tried to do so, white WP page with a message saying I have no access to this page appeared.
9. I had to reopen the page and close the whole Get connected dialog to start editing the page.
10. After this it seems, I have an access to a library even for the Editor account.

So, how this is supposed to work? Can my clients with different WP accounts use my connection to library or not? 

I actually absolutlely don't need them to have this access, in fact I would be much happier, if they were't allowed to access online part of the library at all (just my own saved templates).

But in any way, they shouldn't be asked for a connection, if the admin user is already connected. Definitelly not before they even start to edit the page. Otherwice this would be a VERY annoying "feature".

**Comment by steve231293** [comment]
> Changing column width by typing in the number causes it to only move to the first digit (i.e. 75% is only doing 7%.
> 
> <img alt="Screen Shot 2019-11-27 at 4 22 41 PM" width="1274" src="https://user-images.githubusercontent.com/11357379/69763807-ff276e80-1133-11ea-9a36-5c10f2b689fb.png">
> 
> <img alt="Screen Shot 2019-11-27 at 4 23 05 PM" width="295" src="https://user-images.githubusercontent.com/11357379/69763809-ffc00500-1133-11ea-8dae-8ccb787cdc26.png">

@shilo-ey I have faced with the same problem. could you check it again? [http://prntscr.com/q4u36z](http://prntscr.com/q4u36z)

Theme: Hello
Elementor: Version 2.8.0-beta2

**Comment by sprad001** [comment]
> 
> 
> > Changing column width by typing in the number causes it to only move to the first digit (i.e. 75% is only doing 7%.
> > <img alt="Screen Shot 2019-11-27 at 4 22 41 PM" width="1274" src="https://user-images.githubusercontent.com/11357379/69763807-ff276e80-1133-11ea-9a36-5c10f2b689fb.png">
> > <img alt="Screen Shot 2019-11-27 at 4 23 05 PM" width="295" src="https://user-images.githubusercontent.com/11357379/69763809-ffc00500-1133-11ea-8dae-8ccb787cdc26.png">
> 
> @shilo-ey I have faced with the same problem. could you check it again? http://prntscr.com/q4u36z
> 
> Theme: Hello
> Elementor: Version 2.8.0-beta2

I have experienced this even on the current version.

**Comment by shilo-ey** [comment]
[**@rodolphebertozzo**](https://github.com/rodolphebertozzo) commented on [Dec 1, 2019, 12:42 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-560090437 "2019-12-01T10:42:41Z - Replied by Github Reply Comments"):
> Also remember to add the dark mode here ;)

The dark mode applies to the Editor only, so this screen is outside of the Editor.



[**@knireis**](https://github.com/knireis) commented on [Dec 1, 2019, 11:01 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-560158498 "2019-12-01T21:01:11Z - Replied by Github Reply Comments"):
> Tweak: Moved “Editing Handles” options to the new “Preferences” section
> 
> In 2.8 it does not take the previous setting from 2.7. It just resets itself to the default 'no'. Can you change this so it takes previous setting in account?

This setting has changed from cross-site rule to per-user rule, we will consider inheriting the setting from the previous one, but if a user want to change it from now on, it's much more accessible.


[**@black-eye**](https://github.com/black-eye) commented on [Dec 2, 2019, 12:37 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-560168529 "2019-12-01T22:37:39Z - Replied by Github Reply Comments"):
> One more question:
> How exactly should the Library connection work accross several WP accounts?
> 
> 1. I have Pro acount connected to the email for the main WP Admin account.
> 2. This account seems to be connected to library (at least it's said so on page Elementor > Connect)
> 3. For the first time, when I tried to insert a block from a library, the connection error was thrown.
> 4. After a refresh (which I did as suggested by error message) I was requested to connect the library to an account again.
> 5. After a connection was established, I was finaly able to insert the block from the library
> 6. Then I switched the WP account for an Editor user.
> 7. When I opened a page for editing I was requested to connect to library again (even before the page was opened).
> 8. When I tried to do so, white WP page with a message saying I have no access to this page appeared.
> 9. I had to reopen the page and close the whole Get connected dialog to start editing the page.
> 10. After this it seems, I have an access to a library even for the Editor account.
> 
> So, how this is supposed to work? Can my clients with different WP accounts use my connection to library or not? 
> 
> I actually absolutlely don't need them to have this access, in fact I would be much happier, if they were't allowed to access online part of the library at all (just my own saved templates).
> 
> But in any way, they shouldn't be asked for a connection, if the admin user is already connected. Definitelly not before they even start to edit the page. Otherwice this would be a VERY annoying "feature".

Thanks for testing it. the flow will have the following rules:
1. The Connect process will appear only when a user wants to insert a template from the template library
2. The Connect process will be per user (not per site as the Pro license activation)
3. The user who activated the Pro license is the "Site Owner", and is connected automatically. all the other users will have to connect to authenticate themselves. 



> > Changing column width by typing in the number causes it to only move to the first digit (i.e. 75% is only doing 7%.
> 
> @shilo-ey I have faced with the same problem. could you check it again? http://prntscr.com/q4u36z
> 
> Theme: Hello
> Elementor: Version 2.8.0-beta2


Thanks, now it's much more clear and we could reproduce the issue.


Thanks again to you all!

**Comment by black-eye** [comment]
@shilo-ey 

> 1. The Connect process will appear only when a user wants to insert a template from the template library

So, this means, that experience I had was a bug or obsolete version of connecting?

> 2. The Connect process will be per user (not per site as the Pro license activation)
> 3. The user who activated the Pro license is the "Site Owner", and is connected automatically. all the other users will have to connect to authenticate themselves.

Well, IMHO this will make lots of people ranting, because this "feature" will activelly prevent them to work, how they were used to so far. They could have the feeling, you are actively creating the obstacles to prevent them from fully using the product they have paid for. E PRO is claimed to be fully functional on all buyers sites. Forcing my clients with different user accounts to make (and even pay for) their own Elementor accounts to make library work really entitles to such feeling. :-(

**Comment by PC-Rights** [comment]
Will Elementor be remaining compatible with the WP 4.9.x branch?

This appears at line 51:

`} elseif ( ! version_compare( get_bloginfo( 'version' ), '5.0', '>=' ) ) {`

This would make it unavailable for ClassicPress users. Maybe just an oversight in the code?

**Comment by shilo-ey** [comment]
Hi all!

We just released a new beta version, please update your version to v2.8.0 Beta3.

This beta release includes:

- Tweak: Redesigned colors edit mode with colors reordering in Color Picker
- Tweak: Added Dark Mode theme support to Code control
- Tweak: Removed Library Connect Splash screen from first editor load
- Tweak: Updated "Not Connected" message in Library Connect
- Tweak: Updated `e-icons` library to v5.5.0
- Fix: Option to open links in a new tab has disappeared from WYSIWYG editor (#9735)
- Fix: Resize registers multiple history actions in Column element
- Fix: "Muted" functionality in responsive mode don't apply on background colors
- Fix: Link to Media file in Image widget crashes the editor and throws Uncaught Error
- Fix: Removed redundant Editor Panel Spacer in Image Carousel widget
- Fix: Insufficient privileges to view post after connecting a user below Administrator role to Library Connect 
- Fix: Dark mode glitches
- Fix: Limit transition duration in Ken Burns CSS only to `transform` property
- Fix: Background position in Background Slideshow inherits odd transition duration
- Fix: Right-click menu not being opened when there is no clipboard storage stored


**Please keep this related to the Current beta Only! any other Issue Report or Feature Request feel free to open a new issue.**

Thanks again to everyone.

**Comment by black-eye** [comment]
> Tweak: Redesigned colors edit mode with colors reordering in Color Picker

Nice. Just one issue. When I exit the picker, with Edit mode still open and then open the picker again, I'm unable to drag the swatches. No errors in the console, so I suppose the picker just isn't expecting the Edit mode could be opened again this way. If I click on Done a then Edit again, all works as expected.

**Comment by sermalefico** [comment]
With Beta 3 can´t add custom icons, and can't select my previous custom icons libraries.

**Comment by sprad001** [comment]
The Copy and Paste Style still does not work in Beta 2

**Comment by PC-Rights** [comment]
@shilo-ey  Will Elementor be remaining compatible with the WP 4.9.x branch?

This appears at line 51:

`} elseif ( ! version_compare( get_bloginfo( 'version' ), '5.0', '>=' ) ) {`

This would make it unavailable for [ClassicPress](https://forums.classicpress.net/t/good-bye-elementor/1782) users. Maybe just an oversight in the code?

**Comment by shilo-ey** [comment]
[**@sprad001**](https://github.com/sprad001) commented on [Dec 4, 2019, 2:33 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-561420593 "2019-12-04T00:33:20Z - Replied by Github Reply Comments"):
> The Copy and Paste Style still does not work in Beta 3

Thanks, we will take care of it.


[**@sermalefico**](https://github.com/sermalefico) commented on [Dec 4, 2019, 2:14 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-561415968 "2019-12-04T00:14:16Z - Replied by Github Reply Comments"):
> With Beta 3 can´t add custom icons, and can't select my previous custom icons libraries. 
> 
> Edit- Sorry is a Elementkit conflict problem

Thanks


[**@PC-Rights**](https://github.com/PC-Rights) commented on [Dec 4, 2019, 5:37 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-561462208 "2019-12-04T03:37:50Z - Replied by Github Reply Comments"):
> @shilo-ey  Will Elementor be remaining compatible with the WP 4.9.x branch?
> 
> This appears at line 51:
> 
> `} elseif ( ! version_compare( get_bloginfo( 'version' ), '5.0', '>=' ) ) {`
> 
> This would make it unavailable for [ClassicPress](https://forums.classicpress.net/t/good-bye-elementor/1782) users. Maybe just an oversight in the code?

Thanks, 
Elementor supports the last 3 versions of WordPress (Current - 5.3, 5.2, 5.1, 5.0).
WordPress 4.9 was released on November 16, 2017, and since then it has progressed and evolved in many fields, but most importantly in terms of minimum requirements and new functionality.

We try to align with them and to provide our users with the best experience.

Thanks anyway.

**Comment by lucafagherazzi** [comment]
In the backend, templates do not maintain widget styles.

**Comment by dudaster** [comment]
if you're still there can you fix this age old error?

```
Illegal string offset ‘url’ in
/var/www/html/wp-content/plugins/elementor/includes/base/controls-stack.php
on line 1176 
```
Just add an condition if it's not there don't use it.

**Comment by bainternet** [comment]
We just released a new beta version, please update your version to v2.8 Beta 4.

This beta release includes:

- Tweak: Shapes SVG include: replace `file_exists()` by `is_readable()` (#9734)
- Tweak: Improved Dark Mode styling
- Tweak: Changed SVG Upload icon in Icons control inline skin
- Tweak: UI Changes to Color Picker control 
- Fix: Custom shapes "negative" support fix (#9761)
- Fix: Submit button icon alignment issue in Forms widget
- Fix: Paste style issue in Color Picker
- Fix: Undo/Redo issue in Select2 controls

This Beta is our RC for 2.8


Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

Thanks again to everyone.

**Comment by sprad001** [comment]
I just installed Beta 4 and tested the Copy and Paste Style again. 
What I noticed that the style is actually pastes in, at least the ones I checked, but visually the change cannot be seen in the editor. To see the changes you need to preview it or Exit to Dashboard. Once you get back to the editor then the changes will show up in the editor as well.

**Comment by black-eye** [comment]
> Tweak: UI Changes to Color Picker control
To be honest, I liked the old way better (eith the Edit button). 

Consider this:
How many times will users need to add, delete or rearange colors?
And how many times will they just click on saved color and use it?
For me it's like 5% to 95%. (Well In fact, I define them in the begining, and that's it, but it may differ.)

I would be perfectly fine to use the Edit button for those 5% and leave the picker clean.

And: Imagine the mess when inexperienced users are clicking on + button by mistake, adding and deleting colors randomly or moving them by mistake.

In fact I would be more than happy to disable editing colors (for some user roles) totally. Or disable the editing once the CI colors are set.

**Comment by drawcard** [comment]
Is it also possible to have HEX / RGBA values written in `monospace` font for better readability?

![image](https://user-images.githubusercontent.com/1435638/70293565-94132180-1830-11ea-9cdc-cc4813e0f0f3.png)

Eg. a letter B can look like a number 8 if not using monospace fonts.

**Comment by webfocusdesign** [comment]
> I just installed Beta 4 and tested the Copy and Paste Style again.
> What I noticed that the style is actually pastes in, at least the ones I checked, but visually the change cannot be seen in the editor. To see the changes you need to preview it or Exit to Dashboard. Once you get back to the editor then the changes will show up in the editor as well.

I see the same thing. Paste style seems to do nothing in the editor but when you close it it shows in the page. Tried with a few different widgets and it still doesn't work in beta 4.

**Comment by rodolphebertozzo** [comment]
Great for the color selector! Nice work ;)

**Comment by harrywhitedev** [comment]
How can I add my custom CSS when the Elementor dark mode is enabled? For example I would like to use custom icons as my widgets' icons (consider PNG icons)?

**Comment by MonikaHodkinson** [comment]
When is this due for release please - I am waiting on Ninja Forms to be fixed for nearly a year now .  Thank you

**Comment by dudaster** [comment]
I don't know if it is Beta related but:
**.elementor-button** css rule **text-align** overrides **.elementor-align-left**

**Comment by sprad001** [comment]
After the Beta 4 installation my menu cart display has swapped price with the cart icon and could not adjust spacing between them. With the version control I reinstalled the latest stable version and all came back normal with the price on front and the cart icon on the back. 
Could you please verify if this is happening regularly? The attachment shows how it looked before reverting back to stable version.
![Elementor Menu Cart - Annotation 2019-12-06 215446](https://user-images.githubusercontent.com/58193070/70321822-21c43080-1874-11ea-8fa1-98c0700126c5.jpg)

**Comment by zeinnicholas** [comment]
My only concern is:

How will the data about my library usage be stored? Where? On your servers, or does it mean more entries in wp_postmeta? Can you clarify the database part of this please? I need to understand how this change will affect the storage and performance of databases. Keep in mind there are already other plugins that abuse both wp_options and wp_postmeta like WooCommerce. So how will Elementor store those new informations?

**Comment by shilo-ey** [comment]
[**@sprad001**](https://github.com/sprad001) commented on [Dec 6, 2019, 2:04 PM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-562546625 "2019-12-06T12:04:22Z - Replied by Github Reply Comments"):
> After the Beta 4 installation my menu cart display has swapped price with the cart icon and could not adjust spacing between them. With the version control I reinstalled the latest stable version and all came back normal with the price on front and the cart icon on the back. 
> Could you please verify if this is happening regularly? The attachment shows how it looked before reverting back to stable version.
> Cache clearing and CSS rebuilds were done and did not revert back to normal on the beta.
> ![Elementor Menu Cart - Annotation 2019-12-06 215446](https://user-images.githubusercontent.com/58193070/70321822-21c43080-1874-11ea-8fa1-98c0700126c5.jpg)

Thanks.
 
[**@webfocusdesign**](https://github.com/webfocusdesign) commented on [Dec 6, 2019, 5:45 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-562420571 "2019-12-06T03:45:52Z - Replied by Github Reply Comments"):
> > I just installed Beta 4 and tested the Copy and Paste Style again.
> > What I noticed that the style is actually pastes in, at least the ones I checked, but visually the change cannot be seen in the editor. To see the changes you need to preview it or Exit to Dashboard. Once you get back to the editor then the changes will show up in the editor as well.
> 
> I see the same thing. Paste style seems to do nothing in the editor but when you close it it shows in the page. Tried with a few different widgets and it still doesn't work in beta 4.

It's an issue supposed to be solved in the latest beta, please try clearing your cache with "Empty cache and hard reload"



[**@zeinnicholas**](https://github.com/zeinnicholas) commented on [Dec 8, 2019, 5:14 AM GMT+2](https://github.com/elementor/elementor/issues/9672#issuecomment-562908446 "2019-12-08T03:14:28Z - Replied by Github Reply Comments"):
> My only concern is:
> 
> How will the data about my library usage be stored? Where? On your servers, or does it mean more entries in wp_postmeta? Can you clarify the database part of this please? I need to understand how this change will affect the storage and performance of databases. Keep in mind there are already other plugins that abuse both wp_options and wp_postmeta like WooCommerce. So how will Elementor store those new informations?

Good question,
Elementor won't store any new information besides an authentication token.
This token will be stored in your user's DB (`wp_usermeta` - one line per user), to allow you to access the Template Library.

If you won't connect, no information will be stored.


**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

**Thanks again to everyone.**

**Comment by sprad001** [comment]
> > > I just installed Beta 4 and tested the Copy and Paste Style again.
> > > What I noticed that the style is actually pastes in, at least the ones I checked, but visually the change cannot be seen in the editor. To see the changes you need to preview it or Exit to Dashboard. Once you get back to the editor then the changes will show up in the editor as well.
> > 
> > 
> > I see the same thing. Paste style seems to do nothing in the editor but when you close it it shows in the page. Tried with a few different widgets and it still doesn't work in beta 4.
> 
> It's an issue supposed to be solved in the latest beta, please try clearing your cache with "Empty cache and hard reload"

I have not done the hard reload before but now I did the cache and the hard reload as well but the Style Paste still does not show up in the Elementor editor, just like I reported. The values paste into the widget but no change is showing in the editor. I usually work with Elementor in Firefox, which I did forget to mention. This time I tested it from Chrome as well and got the same result as in Firefox. So the Paste Style only partially works.

**Comment by SniffleValve** [comment]
So far, so good with E 2,8 beta 4 with everything that I have tested :)

**Comment by shilo-ey** [comment]
Elementor v2.8 is now available, Feel free to update.

**Thank you all for your feedback!**

**Comment by webfocusdesign** [comment]
Hi!

FWIW, I'm still seeing the paste style issue (display not updating to pasted style in editor) in the final 2.8.0 in both Chrome and Firefox after emptying browser cache and regenerating CSS styles in Elementor Tools. I have no other caching on in my "sandbox" install where I've been testing 2.8 beta.

**Comment by shilo-ey** [comment]
Can you please elaborate in a new issue? @webfocusdesign

**Comment by pingram3541** [comment]
wordpress.org still showing 2.7.6 as of current time.

With the last beta v4 I am still experiencing issues with the template mechanism importing document settings.  

- When importing a "page" template from the library, regardless if the current editor instance is for a completely brand new page/post or it is an existing one, the prompt to "import document settings" doesn't appear to work.

The page template, background settings and custom css are all missing from the import.

see - https://www.loom.com/share/8226a02119144c70a836083225e912a3

**Comment by webfocusdesign** [comment]
> Can you please elaborate in a new issue? @webfocusdesign

The multiple reports of this here post beta 4 aren't enough? What branch should a new issue be added to since beta 4 is no longer current? Thanks!

**Comment by sprad001** [comment]
> 
> 
> > Can you please elaborate in a new issue? @webfocusdesign
> 
> The multiple reports of this here post beta 4 aren't enough? What branch should a new issue be added to since beta 4 is no longer current? Thanks!

I created one here [https://github.com/elementor/elementor/issues/9811](url). Anyone experiencing the same thing please add your finding here too.

**Comment by zeinnicholas** [comment]
I regret not testing this beta... I understand the choices of changing from RGBA to HEXA, but honestly HEXA is not intuitive in regards of opacity of the color, and it would be OK if the slider showed us the percentage of the opacity while we scroll through the slider, but it doesn't, so we end up having to remember what the value translates to, which in turn slows our workflow. I have no issue in using HEXA just to be clear.

As a side note, this is also odd, since now we have much more space to see the entire RGBA declaration, something that was really annoying before (having to scrub through that tiny input box...).

@shilo-ey do I need to open a new issue to ask you to add the percentage counter to the slider?

**Comment by webcolor-in** [comment]
> > > Can you please elaborate in a new issue? @webfocusdesign
> > 
> > 
> > The multiple reports of this here post beta 4 aren't enough? What branch should a new issue be added to since beta 4 is no longer current? Thanks!
> 
> I created one a few hours ago titled **_Copy and Paste Style Does not show in Editor in Elementor V2.8.0_**. Anyone experiencing the same thing please add your finding there too.
> 
> I have to agree that this problem is there for a while, in fact from Beta 1 but at least now it is pasting the values but the results are not visible in the editor. As a result this problem still brakes the workflow because we cannot see the results.

I am not able to duplicate columns of inner section anymore. Plus not able to copy/paste style of column to other. No doubt, something is wrong. I tried refreshing and checked `console` for js error but nothing was reported there. I can send video cast if required.

**Comment by sprad001** [comment]
@ webceler-in go to here [https://github.com/elementor/elementor/issues/9811](url) and report the copy and paste style issue there. There is a person assigned there to deal with this situation.

**Comment by sainulabidnv** [comment]
Hi, 
How can I delete added colors?. I couldn't see any delete button there.

**Comment by sprad001** [comment]
@sainulabidnv 

> Hi, How can I delete added colors?. I couldn't see any delete button there.

There was a video about. 
You click and drag the color you want to delete,  and as you start to drag a delete box appears below the colors. You release the color in that box and it is deleted.
It is a cool way of deleting the color without possible accidental delete.
